src/sentry/testutils/pytest/sentry.py

```diff
@@ -382,15 +384,17 @@ def _shuffle_d(dct: dict[K, V]) -> dict[K, V]:
     items[:] = new_items
 
 
-def pytest_collection_modifyitems(config: pytest.Config, items: list[pytest.Item]) -> None:
-    """After collection, we need to select tests based on group and group strategy"""
-
-    total_groups = int(os.environ.get("TOTAL_TEST_GROUPS", 1))
-    current_group = int(os.environ.get("TEST_GROUP", 0))
-    grouping_strategy = os.environ.get("TEST_GROUP_STRATEGY", "scope")
-
+def _get_keep_and_discard_items(
```

Comment:
I don't think this separate function actually buys you anything -- other than it now has several boolean traps and is difficult to understand what parameter set it takes

Reply:
The main reason I pulled this into a function was to be able to test it without setting any globals (env variables). I don't think this function has made this code any worse than it was before